src/hooks/useSelect/index.js

```diff
@@ -27,7 +28,10 @@ import {
 } from './utils'
 import * as stateChangeTypes from './stateChangeTypes'
 
-const validatePropTypes = getPropTypesValidator(useSelect, propTypes)
+const validatePropTypes =
+  process.env.NODE_ENV === 'production'
+    ? noop
+    : getPropTypesValidator(useSelect, propTypes)
```

Comment:
I rather recommend to use a condition in `useSelect` as in this case `validatePropTypes` function will be unused and removed:

```jsx
if (process.env.NODE_ENV !== 'production') {
  validatePropTypes(userProps)
}
```